src/Analyzers/CSharp/Analyzers/UseCoalesceExpression/CSharpUseCoalesceExpressionForIfNullStatementCheckDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis.Analyzers.UseCoalesceExpression;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.LanguageService;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageService;
+
+namespace Microsoft.CodeAnalysis.CSharp.Analyzers.UseCoalesceExpression
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal class CSharpUseCoalesceExpressionForIfNullStatementCheckDiagnosticAnalyzer : 
+        AbstractUseCoalesceExpressionForIfNullCheckDiagnosticAnalyzer<
+            SyntaxKind,
+            ExpressionSyntax,
+            StatementSyntax,
+            IfStatementSyntax>
+    {
+        protected override ISyntaxFacts GetSyntaxFacts()
+            => CSharpSyntaxFacts.Instance;
+
+        protected override bool IsNullCheck(ExpressionSyntax condition, [NotNullWhen(true)] out ExpressionSyntax? checkedExpression)
+        {
+            if (condition is BinaryExpressionSyntax(SyntaxKind.EqualsExpression) { Right: LiteralExpressionSyntax(SyntaxKind.NullLiteralExpression) } binary)
+            {
+                checkedExpression = binary.Left;
+                return true;
+            }
+            else if (condition is IsPatternExpressionSyntax { Pattern: ConstantPatternSyntax { Expression: LiteralExpressionSyntax(SyntaxKind.NullLiteralExpression) } } isPattern)
+            {
+                checkedExpression = isPattern.Expression;
+                return true;
+            }
+
+            checkedExpression = null;
+            return false;
+        }
+
+        protected override bool TryGetEmbeddedStatement(IfStatementSyntax ifStatement, [NotNullWhen(true)] out StatementSyntax? whenTrueStatement)
+        {
+            whenTrueStatement = ifStatement.Statement is BlockSyntax { Statements.Count: 1 } block
+                ? block.Statements[0]
+                : ifStatement.Statement;
+
+            return true;
+        }
```

Comment:
This always returns true. Is it intentional to return `BlockSyntax` when it has multiple statements?